### DISCUSSION TITLE
help! I broke Kibana

### DISCUSSION THREAD

**Discussion by bob3rocks** [telekom-security/tpotce#1263] [Q&A] [answered]
Yesterday I inadvertently deleted some files sitting in /data/elk/data/indices/ and today I get this error when viewing Kibana in the GUI:

Kibana server is not ready yet.

Everything else seems to be running and looks good. Cockpit looks good, attack map looks good.

```
[ ========| System |======== ]
     DATE:  Thu 16 Feb 2023 08:01:18 PM UTC
   UPTIME:   20:01:18 up 10:10,  1 user,  load average: 1.93, 2.12, 2.00
    T-POT:  ACTIVE
BLACKHOLE:  DISABLED

NAME                  STATUS                       PORTS
ciscoasa              Up 9 hours                    0.0.0.0:5000->5000/udp, :::5000->5000/udp, 0.0.0.0:8443->8443/tcp, :::8443->8443/tcp
conpot_iec104         Up 9 hours (healthy)          0.0.0.0:161->161/udp, :::161->161/udp, 0.0.0.0:2404->2404/tcp, :::2404->2404/tcp
conpot_ipmi           Up 9 hours (healthy)          0.0.0.0:623->623/udp, :::623->623/udp
conpot_kamstrup_382   Up 9 hours (healthy)          0.0.0.0:1025->1025/tcp, :::1025->1025/tcp, 0.0.0.0:50100->50100/tcp, :::50100->50100/tcp
cowrie                Up 9 hours                    0.0.0.0:22-23->22-23/tcp, :::22-23->22-23/tcp
dicompot              Up 9 hours                    0.0.0.0:11112->11112/tcp, :::11112->11112/tcp
dionaea               Up 9 hours (healthy)          0.0.0.0:135->135/tcp, :::135->135/tcp, 0.0.0.0:445->445/tcp, :::445->445/tcp
elasticpot            Up 9 hours                    0.0.0.0:9200->9200/tcp, :::9200->9200/tcp
elasticsearch         Up 9 hours (healthy)          127.0.0.1:64298->9200/tcp
fatt                  Up 9 hours
honeytrap             Up 9 hours
kibana                Up About a minute (healthy)   127.0.0.1:64296->5601/tcp
log4pot               Up 9 hours                    0.0.0.0:8080->8080/tcp, :::8080->8080/tcp, 0.0.0.0:25565->8080/tcp, :::25565->8080/tcp
logstash              Up 9 hours (healthy)
mailoney              Up 9 hours                    0.0.0.0:25->25/tcp, :::25->25/tcp
map_data              Up 9 hours
map_redis             Up 9 hours
map_web               Up 9 hours                    127.0.0.1:64299->64299/tcp
medpot                Up 9 hours                    0.0.0.0:2575->2575/tcp, :::2575->2575/tcp
nginx                 Up 9 hours
p0f                   Up 9 hours
snare                 Up 9 hours                    0.0.0.0:80->80/tcp, :::80->80/tcp
suricata              Up 9 hours
tanner                Up 9 hours
tanner_api            Up 9 hours
tanner_phpox          Up 9 hours
tanner_redis          Up 9 hours
```

**Comment by t3chn0m4g3** [comment, marked as answer]
You can try to stop T-Pot, find the elkbase.tgz in the `/opt/tpot/etc/objects`folder and then `tar xvfz elkbase.tgz -C /` and then start T-Pot again or reboot.

**Reply by bob3rocks** [reply]
thanks for the fast reply! Trying that now.

**Reply by bob3rocks** [reply]
All services have been online for 15 minutes now, everything looks healthy except that Kibana keeps restarting.

kibana                Up About a minute (healthy)   127.0.0.1:64296->5601/tcp

We browser still says

Kibana server is not ready yet.

**Reply by t3chn0m4g3** [reply]
You can pull up esvue and check if any shards are failing and delete the corresponding index.

**Reply by bob3rocks** [reply]
thanks for that! I found the failing shards and deleted the indexes that were in the red.

**Reply by bob3rocks** [reply]
After deleting the indices that were in the red, I tried restarting tpot, then rebooted, same issue. Kibana won't load and the Kibana PID keeps restarting.

I found this under esview home --> Cluster Health

cluster_name
tpotcluster
status
red

**Reply by t3chn0m4g3** [reply]
At this point you need to check the logs of `docker logs elasticsearch -f` and see what is wrong, do the same for `kibana`. Also `.kibana` index might be corrupt as well. You can start from fresh by stopping T-Pot, deleting `rm -rf /data/elk/`, run the tar command from earlier and start T-Pot again (or reboot).
However I recommend a reinstall before digging any deeper than that.

**Reply by bob3rocks** [reply]
Very well! I've learned some new troubleshooting skills and you're probably right about the reinstall. I will let you know how that goes. Thanks again for the support.

**Reply by bob3rocks** [reply]
docker logs - this looks promising
 current.health="GREEN" message="Cluster health status changed from [YELLOW] to [GREEN] (reason: [shards started

**Reply by bob3rocks** [reply]
![image](https://user-images.githubusercontent.com/23036546/219710668-40a8c9b6-3464-4b84-b4c8-5ca4b1586743.png)

**Reply by bob3rocks** [reply]
Issue appears resolved. Will be more careful with backend data files. Thanks again for the assist!

**Reply by ayisal** [reply]
> At this point you need to check the logs of `docker logs elasticsearch -f` and see what is wrong, do the same for `kibana`. Also `.kibana` index might be corrupt as well. You can start from fresh by stopping T-Pot, deleting `rm -rf /data/elk/`, run the tar command from earlier and start T-Pot again (or reboot). However I recommend a reinstall before digging any deeper than that.

Will deleting data/elk delete the indices?

**Comment by bob3rocks** [comment]
pretty good for 6 hours of data

![image](https://user-images.githubusercontent.com/23036546/219810917-1dea655d-9735-440b-b9af-b5c56f7f0dba.png)

**Comment by bob3rocks** [comment]
I believe the answer to that is yes. That is what I observed.

On Sun, Feb 26, 2023 at 12:54 PM ayisal ***@***.***> wrote:

> At this point you need to check the logs of docker logs elasticsearch -f
> and see what is wrong, do the same for kibana. Also .kibana index might
> be corrupt as well. You can start from fresh by stopping T-Pot, deleting rm
> -rf /data/elk/, run the tar command from earlier and start T-Pot again
> (or reboot). However I recommend a reinstall before digging any deeper than
> that.
>
> Will deleting data/elk delete the indices?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/telekom-security/tpotce/discussions/1263#discussioncomment-5116301>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFPYFATWUAARC25JIT2EUETWZO7G5ANCNFSM6AAAAAAU6TWVRQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
​​​Bob Wachunas​
1(707) 536-1182​
the state of the world really makes me want to {uke}==*::​*
